Framework/Crystal/src/ConvertPeaksWorkspace.cpp

```diff
@@ -0,0 +1,169 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidCrystal/ConvertPeaksWorkspace.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Run.h"
+#include "MantidAPI/Sample.h"
+#include "MantidDataObjects/LeanElasticPeak.h"
+#include "MantidDataObjects/LeanElasticPeaksWorkspace.h"
+#include "MantidDataObjects/PeaksWorkspace.h"
+#include "MantidGeometry/Crystal/OrientedLattice.h"
+#include "MantidGeometry/Instrument/Goniometer.h"
+#include "MantidKernel/Logger.h"
+
+namespace Mantid {
+namespace Crystal {
+
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+/// Config logger
+namespace {
+Logger logger("ConvertPeaksWorkspace");
+}
+
+DECLARE_ALGORITHM(ConvertPeaksWorkspace)
+
+/**
+ * @brief initialization of alg
+ *
+ */
+void ConvertPeaksWorkspace::init() {
+  // Input peakworkspace
+  declareProperty(std::make_unique<WorkspaceProperty<IPeaksWorkspace>>("PeakWorkspace", "", Direction::Input),
+                  "Workspace of Indexed Peaks");
+
+  // donor workspace if going from lean to regular
+  declareProperty(std::make_unique<WorkspaceProperty<IPeaksWorkspace>>("InstrumentWorkpace", "", Direction::Input,
```

Comment:
"InstrumentWorkpace" -> "InstrumentWorkspace"

I think this input should be able to take in any workspace type no just PeaksWorkspace, I should be able to do LoadEmptyInstrument and use that as the InstrumentWorkspace.

Reply:
- The typo will be fixed in the next commit

- My understanding is that the users will need to provide a donor workspace that has the correct settings, which should mean that it is a PeaksWorkspace.  Also, If we were to relax the type restriction, what kind of Workspace type should we use here and what other validations should we do since we throw the type restriction out of the window?

Reply:
Any workspace with an `ExperimentInfo` can have the correct settings. Just make this `WorkspaceProperty<Workspace>` and just check that after doing `ExperimentInfo_sptr inputExperimentInfo = std::dynamic_pointer_cast<ExperimentInfo>(ws)` then check `if (!inputExperimentInfo)` then invalid.

Reply:
The type requirement of `InstrumentWorkspace` is now relaxed to `Workspace`.